### DISCUSSION TITLE
Improve borg output on status bar, and compute directory size/file count with regex patterns

### DISCUSSION THREAD

**Discussion by guslac** [borgbase/vorta#1764] [Ideas]
## Improve borg output

Greetings! for quite some time, I have found the profile-based borg output on Vorta's status bar pretty messy. It gets all mixed up among the child processes, and I can't even know what's going on.
Say we have two profiles named `Default` and `Remote`. If both profiles backup at once, they will have their `[<profile name>] <path>` line on the status bar, and the borg processes will "compete" to show output to the user.

I am guessing it was done this way because the devs didn't expect users to do two or more backups at once, as this is CPU- and IO-intensive, or for some other reason. But that's how I do it, and my old PC handles it well.

So, I suggest a more logical and organized output writing: the currently selected profile must be checked before output is shown, and then one of the borg processes forked by vorta and associated with that profile will be selected to show its output on the status bar. The processes associated with the other profiles will **never** be able to show output unless their respective profile is currently selected by the user on the "Current Profile: " checkbox. So, for example, if `Remote` is the current profile, and `Default` is backing up as well, then only borg output for the `Remote` profile/repo will be shown. This way, it will not be as messy as it gets now.

Another huge issue I see is the Python exception output when something goes wrong. It shouldn't be shown on the status bar, that is not user-friendly. And if it needs to be that way, at least format/ellipsise the exception string to fit on the GUI. In my humble opinion, those exceptions should be written to the vorta log only, not to the status bar.

Another idea: when the user clicks the "log" link (as in `Check the log for details.`), the log file (vorta.log) should be opened with the default text editor. It doesn't make sense to open its containing folder with the default file manager. Also not as practical.

## Compute directory size/file count with regular expression patterns

When the user clicks on the "reload" icon, below where the size and file count are computed for each file and folder added for backup, only simple shell patterns are taken into account.
This pattern: `*.cache/`
and this one: `re:^.*\.cache/`
should have the same effect, and they do have on the borg processes, but they don't on vorta. Vorta will only consider the first pattern, whereas borg, as expected, will properly exclude the .cache dir based on either pattern. This is confusing for the user. As vorta, like borg, is made in Python, it should leverage the `re` module for these filesystem computations with such patterns.

**Comment by real-yfprojects** [comment]
Thank you for your ideas!

Do I get your first point correctly? You suggest to only show status updates for the profile that is currently selected?

Regarding directory size count: Supporting all patterns borg supports basically forces us to rewrite the pattern parser of borg. A solution for this could be calling `borg create --dry-run` but someone has to compare the performance of these two counting methods before this could be implemented.

I will open a feature request for your ideas if you don't beat me to it.

**Reply by guslac** [reply]
Yes, that's correct! I suggested that vorta should show the file backup and error statuses only for the currently selected profile and omit that of the unselected others, if any. So, for instance, `[Default] /home/user/Pictures (A)` should only appear on the status bar when `Default` is selected, and it shouldn't be mixed up with the other ones as it is now. I have 3 profiles by the way.

I understand, and I hope that it will be more performant than it is today, especially for large directories.

Thank you for your consideration!

**Reply by real-yfprojects** [reply]
> Yes, that's correct! I suggested that vorta should show the file backup and error statuses only for the currently selected profile and omit that of the unselected others, if any. So, for instance, `[Default] /home/user/Pictures (A)` should only appear on the status bar when `Default` is selected, and it shouldn't be mixed up with the other ones as it is now. I have 3 profiles by the way.

I had this idea too but there is one problem. How does the user know that a backup of a unselected profile is running in the background. I think such thing should be prominently visible.

**Reply by guslac** [reply]
> > Yes, that's correct! I suggested that vorta should show the file backup and error statuses only for the currently selected profile and omit that of the unselected others, if any. So, for instance, `[Default] /home/user/Pictures (A)` should only appear on the status bar when `Default` is selected, and it shouldn't be mixed up with the other ones as it is now. I have 3 profiles by the way.
> 
> I had this idea too but there is one problem. How does the user know that a backup of a unselected profile is running in the background. I think such thing should be prominently visible.

That's not a problem in my opinion, because vorta already has ways to ensure that, such as the system tray icon change while any backup is running, the notifications for backup start, done, and error, and, of course, even if these were disabled, the user can still switch profiles to see what's going on in each of them, and if an error message is displayed, it won't get overwritten by the output of another profile like it is now, so the user will be able to clearly read it.
